vdj_ann/src/transcript.rs

```diff
@@ -101,7 +122,8 @@ pub fn is_valid(
                 }
                 for stop in jstops.iter() {
                     let n = stop - start;
-                    if pass == 2 || n % 3 == 1 {
+                    // on second pass, go through with checking for stop codon regardless of n % 3 value
+                    if inner_pass == 2 || n % 3 == 1 {
```

Comment:
I think the original code was conflating two checks: (1) full length i.e. having a vstart and jstop and (2) finding a frameshift and/or stop codon. We should split the two checks and specify them as separate fields in the enum